components/scream/src/dynamics/homme/homme_inputs_initializer.cpp

```diff
@@ -0,0 +1,65 @@
+#include "dynamics/homme/homme_inputs_initializer.hpp"
+
+#include "dynamics/homme//interface/scream_homme_interface.hpp"
+#include "dynamics/homme/homme_dimensions.hpp"
+
+#include "Hommexx_Session.hpp"
+#include "ElementsState.hpp"
+#include "Tracers.hpp"
+#include "ElementsForcing.hpp"
+#include "Types.hpp"
+
+namespace scream {
+
+void HommeInputsInitializer::initialize_fields () {
+
+  // We cheat a little bit here. It is way easier for Homme to init
+  // fields during the whole prim initialization sequence.
+  // However, such sequence happens during calls to HommeDynamics'
+  // ctor and initialize method. The AD attempts to init the
+  // inputs *after* all atm process have been inited.
+  // So here's what we do: if the user asks to init inputs,
+  // we simply go ahead and to it, by calling this method inside
+  // HommeDynamics' initialize call. When later the AD calls this
+  // initializer, we simply skip everything, since we already inited
+  // all the fields.
+  // Note: if some other atm proc is trying to init any of the fields
+  //       inited by this class, we the AD should throw an error anyways.
+
+  if (!m_fields_inited) {
+    // Sanity check. This class should init *all* homme's inputs:
```

Comment:
Is there a reason all of this commented code is left in?  Do we plan to revitalize it later on?

Reply:
Ah, good catch. The answer is 'maybe?'... I can remove it for now, and forget about checks like this one. After all, the names for the fields are also going to change (to embrace CF ones), so might as well do everything later. I'll add a TODO comment as a reminder.